src/api/companies.js

```diff
@@ -1,73 +1,83 @@
 const express = require("express");
-const companies = require("./dev/companiesExample");
+const prisma = require("../../prismaClient");
 
 const router = express.Router();
 
-router.get("/", (req, res) => {
-  return res.status(200).json(companies);
+router.get("/", async (req, res) => {
+  try {
+    const company = await prisma.company.findMany();
+    res.status(200).json(company);
+  } catch (err) {
+    res.status(404).json(err);
+  }
 });
 
-router.get("/:id", (req, res) => {
-  const company = companies.find((co) => co.id === +req.params.id);
-  const errorMessage = {
-    message: "Company not found.",
-  };
-
-  return res.status(company ? 200 : 404).json(company || errorMessage);
+router.get("/:id", async (req, res) => {
+  const { id } = req.params;
+  try {
+    const company = await prisma.company.findUnique({
+      where: {
+        id,
+      },
+    });
+    if (company) {
+      res.status(200).json(company);
+    } else {
+      res.status(404).json({ message: "No Company found" });
+    }
+  } catch (err) {
+    res.status(404).json(err);
+  }
 });
 
-router.post("/", (req, res) => {
-  const { name, zip_code, city } = req.body;
-  const errorMessage = { message: "Missing fields." };
-
-  if (!name || !zip_code || !city) {
-    return res.status(400).json(errorMessage);
+router.post("/", async (req, res) => {
+  const { name, logoUrl } = req.body;
+  try {
+    const company = await prisma.company.create({
+      data: {
+        name,
+        logoUrl,
+      },
+    });
+    res.status(201).json(company);
+  } catch (err) {
+    res.status(404).json(err);
   }
-
-  companies.push({
-    id: companies.length + 1,
-    name,
-    zip_code,
-    city,
-    created_at: "01-01-2000",
-    updated_at: "02-02-2000",
-  });
-
-  res.status(201).json(companies[companies.length - 1]);
 });
 
-router.put("/:id", (req, res) => {
-  const { name, zip_code, city } = req.body;
-  const index = companies.findIndex((company) => company.id === +req.params.id);
-  const errorMessage = { message: "Company not found." };
+router.put("/:id", async (req, res) => {
+  const { id } = req.params;
+  const { name, logoUrl } = req.body;
 
-  if (index === -1) {
-    return res.status(404).json(errorMessage);
+  if (!name || !logoUrl) {
```

Comment:
Not useful in put route. If undefined, prisma don't update field. This checks have to be in POST logic (logoUrl is nullable).